ApiHelper.php

```diff
@@ -138,35 +138,32 @@ public function getMonthlyPageviews( $pages, $start, $end ) {
 					$titles[] = $r['title'];
 				}
 			}
-			// Get monthly pageviews for all of the titles i.e. original page and its redirects
+			// Get monthly pageviews for all of the titles i.e. original page and its redirects using promises
+			$promises = [];
 			foreach ( $titles as $title ) {
 				$url = 'https://wikimedia.org/api/rest_v1/metrics/pageviews/per-article/en.wikipedia/all-access/user/' . rawurlencode( $title ) . '/monthly/' . $start . '/' . $end;
-				$result = json_decode( @file_get_contents( $url ), true );
-				if ( isset( $result['items'] ) ) {
-					$results[$page] += (int)$result['items'][0]['views'];
-				} else {
-					// Report missing data
-					/*
+				$promise = $client->getAsync( $url );
+				$promises[] = $promise; // Add the promise for each request to promises array
+			}
+			try {
+				$responses = GuzzleHttp\Promise\settle( $promises )->wait();
+			} catch ( Exception $e ) {
+				// Ignore since we are logging those below anyway.
```

Comment:
The foreach will give a warning, but won't fatal. That's probably a decent outcome. The warning will be logged in the error.log, but the script will keep running for the next project. It might be better to double-check $responses before the foreach and explicitly record an error in the log.txt.